tests/functional/restart/53-task-prerequisites.t

```diff
@@ -28,8 +28,8 @@ install_suite "${TEST_NAME_BASE}" "${TEST_NAME_BASE}"
 
 run_ok "${TEST_NAME_BASE}-validate" cylc validate "${SUITE_NAME}"
 
-suite_run_ok "${TEST_NAME_BASE}-run" cylc run "${SUITE_NAME}" --stop-point=2
-poll_suite_stopped
+suite_run_fail "${TEST_NAME_BASE}-run" cylc run "${SUITE_NAME}" --stop-point=2 --no-detach
+
```

Comment:
Why should this fail now?

Reply:
Because the flow is now being run with `--no-detach` which means that `cylc run` is run in the foreground until the flow finishes. It then exits 0/1 based on the task pool.

Before the flow was being run in the background so `cylc run` always exited 0 as long as it succeeded to launch the flow.

The background approach was flaky as the log file and DB are closed after `poll_suite_stopped` returns.

Reply:
Oh I see, and the workflow in question exits with 1 because of the deliberate failed task that causes it to stall